source/common/router/router.cc

```diff
@@ -694,6 +704,36 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
       *this, std::move(generic_conn_pool), can_send_early_data, can_use_http3);
   LinkedList::moveIntoList(std::move(upstream_request), upstream_requests_);
   upstream_requests_.front()->acceptHeadersFromRouter(end_stream);
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.streaming_shadow")) {
```

Comment:
might make sense to latch this for a given FC e.g. for the router, what happens if it gets flipped halfway through a stream?

Reply:
Done; latched on router construction.